packages/core/src/tools/limitModification.ts

```diff
@@ -18,7 +19,7 @@ export function limitModification<T extends Context, Result>(
     const originalType = getType(originalValue)
     const newType = getType(newValue)
     if (newType === originalType) {
-      set(object, path, newValue)
+      set(object, path, sanitize(newValue))
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏it could be interesting to add a test for this one

Reply:
👌